src/main/java/org/opensearch/geospatial/action/upload/geojson/Uploader.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.geospatial.action.upload.geojson;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.StepListener;
+import org.opensearch.action.bulk.BulkItemResponse;
+import org.opensearch.action.bulk.BulkRequestBuilder;
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+
+/**
+ * Uploader will upload GeoJSON objects from UploadGeoJSONRequestContent as
+ * Documents to given index in three stage.
+ * At first stage (preUpload), resources like index, mapping, pipeline with
+ * GeoJSON Feature processors, will be created.
+ * At second stage (upload), Feature will be extracted from GeoJSON and indexed using
+ * BulkAction. This supports both Feature and FeatureCollection.
+ * At third stage (postUpload), previously created pipeline will be deleted
+ * At final stage response or failure will be added to the listener.
+ */
+public class Uploader {
+
+    public static final String ERROR_MESSAGE_DELIMITER = "\n";
+
+    private static final Logger LOGGER = LogManager.getLogger(Uploader.class);
+
+    private final IndexManager indexManager;
+    private final PipelineManager pipelineManager;
+    private final ContentBuilder contentBuilder;
+
+    public Uploader(IndexManager indexManager, PipelineManager pipelineManager, ContentBuilder contentBuilder) {
+
+        this.indexManager = Objects.requireNonNull(indexManager, "IndexManager instance cannot be null");
+        this.pipelineManager = Objects.requireNonNull(pipelineManager, "PipelineManager instance cannot be null");
+        this.contentBuilder = Objects.requireNonNull(contentBuilder, "ContentBuilder instance cannot be null");
+    }
+
+    /**
+     * upload abstracts following operations from request
+     * 1. Create index if it doesn't exist.
+     * 2. Create pipeline with {@link org.opensearch.geospatial.processor.FeatureProcessor}
+     * 3. Prepare Content from {@link UploadGeoJSONRequestContent#getData()}
```

Comment:
What would a user do if their content doesnt fit in a single request?

Reply:
Having geojson file content exceeds max http content size is not practical given it is more of auxiliary data than domain data. We can recommend users to split the file into multiple files and call PUT API.
I created an issue for exceeding BulkApi limit #36 .

Reply:
Do you have a sample of what a user might upload, so I get a better idea of size?

Reply:
http://www.naturalearthdata.com/downloads/10m-cultural-vectors/